### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by j-f1**
- switch to TypeScript
- make the CSS better match the default dictionary
- parse parts of speech and expand them (since we don’t have space constraints in an electronic dictionary)
  - I tried to piece these together by referencing the words that used the abbreviations and plugging the abbreviations into Wiktionary, but I’m not entirely confident with all of them so if you have some knowledge in this area I’d appreciate a once-over on that
- add a couple missing abbreviations
- add support for matrices (which are only used once or twice, but still)

I’m happy to revert some of the more dramatic changes (TypeScript and ES6-ification specifically) if you’d like.

### CODE REVIEW

**Comment by j-f1** [comment]
Also, a quick question: It looks like this repo is licensed under the GPL, but the package.json metadata says the license is ISC (which is like MIT). Which license do you intend to use?

File: package.json
Comment (jeffbyrnes) [inline package.json]: Probably good to leave the `name`, `version`, etc. in here, since it’s useful metadata (even if this is not published as a package like most modules).

**Comment by jeffbyrnes** [comment]
Can’t say about the licensing, but I do like this overall! I’m not well-versed in TypeScript, but I hear good things.

That said, it _might_ be more beneficial to keep things simpler & not introduce TS? I’ll defer to @ponychicken as the maintainer for sure.

Also, #10 merged recently, so this needs a rebase.

File: package.json
Comment (j-f1) [reply package.json]: I removed it because all the metadata is visible elsewhere and just clutters up package.json.

**Comment by j-f1** [comment]
Rebased :+1:

File: partOfSpeech.ts
Comment (ponychicken) [inline partOfSpeech.ts]: I think the additional explaination in the braces (simple past tense) can be left out. It seems to be used in the form imp. & p. p. a lot.

File: partOfSpeech.ts
Comment (ponychicken) [inline partOfSpeech.ts]: this most likely was meant to be imp. & p. p.

Comment (ponychicken) [reply partOfSpeech.ts]: so susbtition should be "imperfect & past participle"

File: partOfSpeech.ts
Comment (ponychicken) [inline partOfSpeech.ts]: For my taste it goes a bit over on the balance between brevity <-> understandibility.
I think "tense" could be dropped in these lines

File: partOfSpeech.ts
Comment (ponychicken) [inline partOfSpeech.ts]: i think its ok to drop "case" here

File: template/Info.plist
Comment (ponychicken) [inline template/Info.plist:18]: What does this do?

**Review by ponychicken — CHANGES_REQUESTED** [review]
Thanks!
I haven't used Typescript before, but it seems uncomplicated enough to try.
I added some comments. I haven't had a chance to check out the CSS changes yet.

I fixed the lint scripts on master, and all the tests should be passing now...

File: template/Info.plist
Comment (j-f1) [reply template/Info.plist:18]: It lets you go to Go → Front/Back Matter → Webster’s 1913 and see the things that weren’t placed under a word.

**Comment by j-f1** [comment]
Getting this error in the lint script:
```
Error Error: Received status code 403
    at IncomingMessage.res.on (/action/lib/request.js:13:25)
    at IncomingMessage.emit (events.js:203:15)
    at endReadableNT (_stream_readable.js:1145:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
{"message":"Resource not accessible by integration","documentation_url":"https://developer.github.com/v3/checks/runs/#create-a-check-run"}
```

**Comment by ponychicken** [comment]
The lint thing needs some love: https://github.com/hallee/eslint-action
Only recently added support for .ts files, and right now we are using v1.0 which doesnt include this patch. But the master branch is also broken :( 
I guess we either fork & fix it, or take it out.

**Comment by ponychicken** [comment]
Bildschirmfoto 2020-05-20 um 08 59 26
Bildschirmfoto 2020-05-20 um 09 02 09

I added the custom font to fix the display of the accents (see screenshot). But maybe there are also other ways to fix it without adding a large font...

**Comment by ponychicken** [comment]
But the _adjective_ part looks really nice!

**Comment by jeffbyrnes** [comment]
I think we can pull out the automated linting for now.

**Comment by j-f1** [comment]
Weird, here’s what I’m seeing:
Screenshot_2020-05-20 08 11 06

**Comment by jeffbyrnes** [comment]
I will say, for me, I’ve really enjoyed that the Webster’s formatting is different. I.e., the red serif typeface for the entry titles.

**Comment by j-f1** [comment]
I could try to add a setting in Dictionary preferences to switch the font.

Also, I tried to hide etymology from the dictionary popup you get by force clicking a word, but it seems to have stopped working.

**Comment by ponychicken** [comment]
Thanks – i merged most of you changes, except the CSS

**Pull request closed without merging**
